devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -112,6 +121,22 @@ public void execute(Task test) {
                 JavaPlugin.class,
                 javaPlugin -> {
                     project.afterEvaluate(this::afterEvaluate);
+                    JavaCompile compileJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_JAVA_TASK_NAME);
+                    compileJavaTask.mustRunAfter(quarkusPrepare);
+                    quarkusPrepare.setSourceRegistrar(compileJavaTask::source);
+
+                    try {
+                        // TODO: support kotlin
+                        tasks.getByName("compileTestKotlin");
+                    } catch (UnknownTaskException noKotlin) {
+                        JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
+                        compileTestJavaTask.dependsOn(quarkusPrepareTests);
+                        quarkusPrepareTests.setSourceRegistrar(compileTestJavaTask::source);
+                    }
+
+                    Path projectPath = project.getProjectDir().toPath();
+                    quarkusPrepare.setSourcesDirectory(projectPath.resolve("src").resolve("main"));
+                    quarkusPrepareTests.setSourcesDirectory(projectPath.resolve("src").resolve("test"));
```

Comment:
If you look below you'll see SourceSet's. That'll be the proper way to get the sources dirs.

Reply:
parent of java's source works too :)